runners/flink/flink_runner.gradle

```diff
@@ -177,9 +177,17 @@ dependencies {
   if (flink_version.startsWith('1.10')) {
       compile "org.apache.flink:flink-connector-kafka-0.11_2.11:1.10-lyft20200508"
       compile "com.lyft:streamingplatform-kinesis:2.1-SNAPSHOT"
+      compile("com.lyft:streamingplatform-kafka:2.1-SNAPSHOT") {
+        exclude group: "org.apache.flink", module: "flink-connector-kafka-0.11_2.11"
```

Comment:
Please add a comment about why the excludes are necessary.

Reply:
I added one exclusion as there was an explicit reference two lines above.  https://github.com/lyft/beam/blob/5db789871605c0406bce6a881b520fe37cdc9514/runners/flink/flink_runner.gradle#L178

Reply:
Is it really necessary to exclude these? Wouldn't Gradle anyways include the kafka connector due to the reference above? Also, `flink-streaming-java_2.11` will be included as part of the other dependencies.

Reply:
Please check what dependencies get pulled in via com.lyft:streamingplatform-kafka:2.1-SNAPSHOT and how that changes the job server jar size. I seem to have noticed increased bloating first when upgrading Beam and then also when upgrading Flink.

Reply:
These two were the only dependencies that were being pulled and as they were being included through others, i made the explicit exclusion.   

If you observe, the version we have made explicit for flink < 1.10 is  `"org.apache.flink:flink-connector-kafka-0.11_2.11:1.8-lyft20190924"`   which isn't what comes from `com.lyft:streamingplatform-kafka:1.4-SNAPSHOT`

 +--- com.lyft:streamingplatform-kafka:1.4-SNAPSHOT
|    |    +--- org.apache.flink:flink-connector-kafka-0.11_2.11:1.8-lyft20200316 (*)
|    |    \--- org.apache.flink:flink-streaming-java_2.11:1.8-lyft20200316 -> 1.8.3 (*)

Reply:
As discussed in person, the above dependency graph is invalid. Flink runner 1.10 depends on Flink 1.10 and SP 2.1

Reply:
I specify the group and module only in the exclusion config and purposefully exclude `version`.

On Flink 1.10, I see 

\--- com.lyft:streamingplatform-kafka:2.1-SNAPSHOT
          +--- **org.apache.flink:flink-connector-kafka-0.11_2.11**:1.10-lyft20200427 -> 1.10-lyft20200508 (*)
          \--- **org.apache.flink:flink-streaming-java_2.11**:1.10-lyft20200427 -> 1.10.0 (*)

Reply:
It looks like the case of older, locally cached snapshot dependency. We can leave this as is since it is working.